psydac/feec/derivatives.py

```diff
@@ -319,21 +319,26 @@ def tosparse(self, **kwargs):
 
             if self._diffdir == d:
                 # if we are at the differentiation direction, construct differentiation matrix
-                maindiag = np.ones(domain_local) * (-sign)
-                adddiag = np.ones(domain_local) * sign
-
-                # handle special case with not self.domain.parallel and not with_pads and periodic
-                if self.domain.periods[d] and not self.domain.parallel and not with_pads:
-                    # then: add element to other side of the array
-                    adddiagcirc = np.array([sign])
-                    offsets = (-codomain_local+1, 0, 1)
-                    diags = (adddiagcirc, maindiag, adddiag)
+                if codomain_local == 1 and domain_local == 1 and self.domain.periods[d]:
+                    # case of one cell and periodic BC should be treated as a constant
+                    addmatrix = spa.coo_array((codomain_local, domain_local))
```

Comment:
why call this `addmatrix` since it will be a term in a Kronecker _product_ ?

Reply:
(also for clarity, you can add a comment specifying that this is a zero matrix)

Reply:
This is the name that was used for this variable before. I can change it though. Any suggestion?

Reply:
why not `directional_matrix` ?